### ISSUE TITLE
models.js ensure that required variables are submitted

### ISSUE THREAD

**Issue by kuanb** [slco-2016/clientcomm#147]
In `models.js`, we do not reject when required variables are not supplied. 

If any variables are `undefined` at the execution of promise, we should cancel operation and run `reject()`.

**Comment by kuanb** [comment]
Example? https://github.com/slco-2016/clientcomm/blob/master/models/classes/group.js

**Comment by maxmcd** [comment]
Ah, model validators. I feel like there's more we can leverage from JS here to catch undefined parameters. `arguments` provides an array of all arguments passed to the function correct? And if we can abstract that maybe there's a way to not implement this manually in every function. Might want to look into class inheritance and see if we can throw in some kind of call to `super` although I'm thinking with python here and not sure how we would do that with ES6 classes. Love the idea, but yeah we might be able to DRY up the implementation.

**Comment by kuanb** [comment]
cool

**Comment by kuanb** [comment]
the other side would be to assume that everything is good when it is input into the class funcs... tho i think it might be poor form?

**Comment by maxmcd** [comment]
Yeah, I like that it's loud and angry about params. When I implemented the create function for `Message` I ended up throwing a params object in because the lack of named parameters got confusing (maintaining order, remember which param is which, etc...). Maybe the create functions take an object and we reference that against a list of attributes stored on the model? Then we might be able to abstract away the create function to a parent class entirely? I can work on a proof of concept of that. Also feels like we might be reinventing the wheel here, maybe there's some material out there for people trying to force Python/Ruby-style models on node.js.

**Comment by kuanb** [comment]
This is still relevant - while we do have base models, we need to incorporate some sort of form validation and if it is to be an extension of base model, then that sounds good.
